File: pyiArduinoI2Cmatrix/Cpp/Image.cpp
// Данный пример выводит изображения myImage_1 и myImage_2.
#include "../iarduino_I2C_Matrix_8x8.h"
//#include <cstdint>
#include <cstdlib>
#include <iostream>
typedef unsigned char byte;
                                                                 // Создаём изображение "смайлик".
byte myImage_1[8] = { 0b00111100,                                //                        ####
                      0b01000010,                                //                       #    #
                      0b10100101,                                //                      # #  # #
                      0b10000001,                                //                      #      #
                      0b10100101,                                //                      # #  # #
                      0b10011001,                                //                      #  ##  #
                      0b01000010,                                //                       #    #
                      0b00111100 };                              //                        ####
                                                                 // Создаём изображение "телевизор".
byte myImage_2[8] = { 0b01000100,                                //                       #   #
                      0b00101000,                                //                        # #
                      0b00010000,                                //                         #
                      0b11111111,                                //                      ########
                      0b10000011,                                //                      #     ##
                      0b10000011,                                //                      #     ##
                      0b10000011,                                //                      #     ##
                      0b11111111 };                              //                      ########
iarduino_I2C_Matrix_8x8 disp;

void loop(void);

int main(int argc, char** argv)
{
    if(argc > 1) {
	    disp.changeBus(argv[1]);
    }
    delay(500);
    disp.begin();
    for (;;) {
	    loop();
    }
}

void loop()
{
	disp.drawImage(myImage_1);
	delay(2000);
	disp.drawImage(myImage_2);
	delay(2000);
}

File: ino/py/Image.py
# Данный пример выводит изображения myImage_1 и myImage_2.      #
                                                                 # Создаём изображение "смайлик".
myImage_1 = [ 0b00111100,                                #                        
                      0b01000010, #* Посмотрите, изображение */  #                           
                      0b10100101, #* можно увидеть прямо в   */  #                          
                      0b10000001, #* коде!!!                 */  #                            
                      0b10100101,                                #                          
                      0b10011001,                                #                          
                      0b01000010,                                #                           
                      0b00111100 ]                               #                        
                                                                 # Создаём изображение "телевизор".
myImage_2 = [ 0b01000100,                                #                            
                      0b00101000,                                #                            
                      0b00010000,                                #                             
                      0b11111111,                                #                      
                      0b10000011,                                #                           
                      0b10000011,                                #                           
                      0b10000011,                                #                           
                      0b11111111 ]                               #                      
from pyiArduinoI2Cmatrix import *                               # Подключаем библиотеку для работы с LED матрицей 8x8.
import sys
from time import sleep
disp = pyiArduinoI2Cmatrix(int(sys.argv[1]))                     # Объявляем объект disp для работы с LED матрицей 8x8, указывая её адрес на шине I2C.
sleep(.5)
disp.begin()
#sleep(.5)
#disp.fillScr()
sleep(.5)
disp.clrScr()
                                                                 #
                                                                 #
while True:                                                      #
    disp.drawImage(myImage_1),  sleep(2)                       # Выводим на дисплей изображение массива myImage_1 и ждём пару секунд.
    disp.drawImage(myImage_2),  sleep(2)                       # Выводим на дисплей изображение массива myImage_2 и ждём пару секунд.
